src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/AddDocumentController.kt

```diff
@@ -32,22 +34,29 @@ class AddDocumentController(
   fun addDocument(
     @PathVariable("recallId") recallId: RecallId,
     @RequestBody body: AddDocumentRequest
-  ): ResponseEntity<AddDocumentResponse> {
+  ): ResponseEntity<AddDocumentResponse> =
     // TODO:  Restrict the types of documents that can be uploaded. i.e. RECALL_NOTIFICATION, REVOCATION_ORDER
-    val documentId = try {
-      recallDocumentService.uploadAndAddDocumentForRecall(
-        recallId = recallId,
-        documentBytes = body.fileContent.toBase64DecodedByteArray(),
-        documentCategory = body.category,
-        fileName = body.fileName
-      )
-    } catch (e: RecallNotFoundException) {
-      throw ResponseStatusException(BAD_REQUEST, e.message, e)
+    uploadDocument(recallId, body).map { documentId ->
+      ResponseEntity
+        .created(URI.create("$baseUri/recalls/$recallId/documents/$documentId"))
+        .body(AddDocumentResponse(documentId = documentId))
+    }.recover {
+      ResponseEntity.badRequest().build()
     }
 
-    return ResponseEntity
-      .created(URI.create("$baseUri/recalls/$recallId/documents/$documentId"))
-      .body(AddDocumentResponse(documentId = documentId))
+  private fun uploadDocument(
+    recallId: RecallId,
+    body: AddDocumentRequest
+  ) = try {
+    recallDocumentService.scanAndStoreDocument(
+      recallId = recallId,
+      documentBytes = body.fileContent.toBase64DecodedByteArray(),
+      documentCategory = body.category,
+      fileName = body.fileName
+    )
+  } catch (e: RecallNotFoundException) {
+    // Anyone object ot getting rid of this and letting the default NotFoundException return NOT_FOUND?
```

Comment:
I can see some minor benefit to the BAD_REQUEST response, but happy for it to go too.  I don't see a justification for custom exceptions in this code.